numba/parfors/parfor.py

```diff
@@ -3566,15 +3566,17 @@ def get_reduction_init(nodes):
     Get initial value for known reductions.
     Currently, only += and *= are supported.
     """
-    require(len(nodes) >=1)
-    # there could be an extra assignment after the reduce node
+    require(len(nodes) >= 1)
+    # there could be multiple extra assignments after the reduce node
     # See: test_reduction_var_reuse
-    if isinstance(nodes[-1].value, ir.Var):
-        require(len(nodes) >=2)
-        require(nodes[-2].target.name == nodes[-1].value.name)
-        acc_expr = nodes[-2].value
-    else:
-        acc_expr = nodes[-1].value
+    cur_index = 1
+    acc_expr = nodes[-cur_index].value
+    while isinstance(acc_expr, ir.Var):
+        require(len(nodes) >= cur_index + 1)
+        require(nodes[-(cur_index + 1)].target.name == nodes[-cur_index].value.name)
+        cur_index += 1
+        acc_expr = nodes[-cur_index].value
```

Comment:
It seems like this might be relying on a naming pattern? The following variant on the original MWR breaks for variable `t`:
```python
from numba import njit, prange
import numpy as np

@njit(parallel=True)
def test_impl(A, cond):
    s = 1
    t = 10
    for i in prange(A.shape[0]):
        if cond[i]:
            s += 1
            t += 1
        else:
            s += 2
    return s, t

print(test_impl(np.ones(10), (np.arange(10) % 2).astype('bool')))
```